src/app/share/views/SharedGuestSignUp/ShareGuestSingUpView.tsx

```diff
@@ -0,0 +1,306 @@
+  const [signupError, setSignupError] = useState<Error | string>();
+  const [showError, setShowError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const { doRegisterPreCreatedUser } = useSignUp('activate');
+  const [passwordState, setPasswordState] = useState<{
+    tag: 'error' | 'warning' | 'success';
+    label: string;
+  } | null>(null);
+  const [invitationId, setInvitationId] = useState<string>();
+  const [showPasswordIndicator, setShowPasswordIndicator] = useState(false);
+  const user = useSelector((state: RootState) => state.user.user) as UserSettings;
+  const mnemonic = localStorageService.get('xMnemonic');
+
+  const [invitationValidation, setInvitationValidation] = useState({
+    isLoading: true,
+    isValid: false,
+  });
+
+  const formInputError = Object.values(errors)[0];
+
+  let bottomInfoError: null | string = null;
+
+  if (formInputError?.message) {
+    bottomInfoError = formInputError.message;
+  } else if (showError && signupError) {
+    bottomInfoError = signupError.toString();
+  }
+
+  const validateInvitation = async (id: string) => {
+    setInvitationValidation((prev) => ({ ...prev, isLoading: true }));
+
+    try {
+      await shareService.validateSharingInvitation(id);
+      setInvitationId(id);
+      setInvitationValidation((prev) => ({ ...prev, isLoading: false, isValid: true }));
+    } catch (error) {
+      errorService.reportError(error);
+      setInvitationValidation((prev) => ({ ...prev, isLoading: false, isValid: false }));
+    }
+  };
+
+  useEffect(() => {
+    const inviteId = qs.invitation as string;
+
+    if (inviteId) {
+      validateInvitation(inviteId);
+    } else {
+      return navigationService.push(AppView.Signup);
+    }
+  }, [invitationId]);
+
+  useEffect(() => {
+    if (user && mnemonic) {
+      dispatch(userActions.setUser(user));
+      if (user?.registerCompleted && mnemonic) {
+        return navigationService.push(AppView.Shared);
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    if (password.length > 0) onChangeHandler(password);
+  }, [password]);
+
+  function onChangeHandler(input: string) {
+    setIsValidPassword(false);
+    if (input.length > MAX_PASSWORD_LENGTH) {
+      setPasswordState({ tag: 'error', label: translate('modals.changePasswordModal.errors.longPassword') });
+      return;
+    }
+
+    const result = testPasswordStrength(input, (qs.email as string) === undefined ? '' : (qs.email as string));
+
+    setIsValidPassword(result.valid);
+    if (!result.valid) {
+      setPasswordState({
+        tag: 'error',
+        label:
+          result.reason === 'NOT_COMPLEX_ENOUGH'
+            ? 'Password is not complex enough'
+            : 'Password has to be at least 8 characters long',
+      });
+    } else if (result.strength === 'medium') {
+      setPasswordState({ tag: 'warning', label: 'Password is weak' });
+    } else {
+      setPasswordState({ tag: 'success', label: 'Password is strong' });
+    }
```

Comment:
Could you add a TODO in order to remember extract this? It is repeated in a few components.